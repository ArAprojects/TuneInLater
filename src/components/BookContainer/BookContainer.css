.BookContainer_section {
  background-color: lightblue;
  display: flex;
  flex-direction: column;
}
