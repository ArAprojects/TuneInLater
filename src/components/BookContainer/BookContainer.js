import React from 'react';
import Book from '../Book/Book';
import './BookContainer.css';

const BookContainer = ({ audiobooks, toggleFavorite }) => {
  let audiobookList = audiobooks.map((book, index) => {
    return <Book key={index} bookName={book.bookName} author={book.author} art={book.art} toggleFavorite={toggleFavorite} id={book.id} favorite={book.favorite} />
  })

  return (
    <section className="BookContainer_section">
      {audiobookList}
    </section>
  )

}

export default BookContainer;
