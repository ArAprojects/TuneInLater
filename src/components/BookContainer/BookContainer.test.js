import React from 'react';
import { shallow } from 'enzyme';
import BookContainer from './BookContainer';

describe('BookContainer', () => {
  let wrapper;
  const mockAudioBooks = [{
    artwork_url: 'http//:image.png',
    author_name: 'Dr. Seuss',
    book_id: 1,
    book_name: 'The Cat in the Hat',
    description: 'Story Description',
    favorite: true,
    primary_genre_name: 'children',
    release_date: '04/23/1990'
  }];
  const mockToggleFavorite = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<BookContainer 
      audiobooks={mockAudioBooks}
      toggleFavorite={mockToggleFavorite}
    />)
  })

  it('should keep an up-to-date snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
