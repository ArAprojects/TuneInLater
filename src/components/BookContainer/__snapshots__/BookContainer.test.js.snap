// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContainer should keep an up-to-date snapshot 1`] = `
<section
  className="BookContainer_section"
>
  <Book
    artwork_url="http//:image.png"
    author_name="Dr. Seuss"
    book_id={1}
    book_name="The Cat in the Hat"
    description="Story Description"
    favorite={true}
    key="0"
    primary_genre_name="children"
    release_date="04/23/1990"
    toggleFavorite={[MockFunction]}
  />
</section>
`;
